src/kudu/consensus/leader_election.cc

```diff
@@ -802,27 +807,15 @@ void FlexibleVoteCounter::AppendPotentialLeaderUUID(
   bool quorum_satisfied = true;
   bool quorum_satisfaction_possible = true;
 
-  if (FLAGS_srd_strict_leader_election_quorum) {
```

Comment:
Moving this check to one place in `AreAllMajoritiesSatisfied()`. However, this means that a potential leader is now only based on intersection with the previous leader, we don't take into consideration special majorities like `FLAGS_srd_strict_leader_election_quorum` and `FLAGS_include_candidate_region` etc. which I think is fine since we're only making the case more pessimistic.